src/spec.rs

```diff
@@ -41,16 +52,14 @@ pub(crate) fn commit_ivk(
     // We rely on the API contract that to_le_bits() returns at least PrimeField::NUM_BITS
     // bits, which is equal to L_ORCHARD_BASE.
     let domain = sinsemilla::CommitDomain::new(&"z.cash:Orchard-CommitIvk");
-    let ivk = domain.short_commit(
-        iter::empty()
-            .chain(ak.to_le_bits().iter().by_val().take(L_ORCHARD_BASE))
-            .chain(nk.to_le_bits().iter().by_val().take(L_ORCHARD_BASE)),
-        rivk,
-    );
-
-    // Convert from pallas::Base to pallas::Scalar. This requires no modular reduction
-    // because Pallas' base field is smaller than its scalar field.
-    pallas::Scalar::from_repr(ivk.to_repr()).unwrap()
+    mod_r_p(
+        domain.short_commit(
```

Comment:
There's a slight wrinkle here because `short_commit` can fail (with negligible probability) due to incomplete additions in Sinsemilla. For the time being just document that as a to-do.